doc/y_cable/design_doc.md

```diff
@@ -0,0 +1,226 @@
+## Multiple y cable vendor API design
+
+### Overview
+
+This document summarizes the approach taken to accommodate multiple
+Y cable vendors for supporting dual-ToR configurations in SONiC.
+All the design choices are presented
+such that vendors can implement the Y cable interface with ease and effeciency
+and all the feature requirements are met.
+
+
+Challenge: to provide an interface for Y cable to interact with PMON docker and HOST
+           which can be uniform across all vendors.
+
+### Directory Structure
+
+#### Background
+
+- We need to have an appropriate place where vendors can implement their modules/packages and provide Sonic with an abstraction layer where pmon and host can access the Y cable packages of all the vendors with ease
+   
+  - Currently the assumption is Y cable package implementation would be in python itself
+  - If a vendor provides shared library, that scenario would be considered accordingly
+  - Since sonic_y_cable (sonic-platform-common/sonic_y_cable) is already built as a package today (only for a single Y cable vendor with sonic-buildimage) vendors can also build packages along similar lines.
```

Comment:
A little confused here. Unlike sonic-platform, where platform vendors build a package, here we intend for the sonic_y_cable package to hold all cable vendors' implementations. What do you mean by "vendors can also build packages along similar lines"? Do you intend for each vendor to build a package?

For simplicity, I thought we could simply import the appropriate Python file based on the vendor+part number combo, vendors would not provide "packages" per se.

Reply:
removed the package, vendors just create their modules under the vendor name directory